src/Raven.Server/Documents/Handlers/Processors/Attachments/RangeHelper.cs

```diff
@@ -0,0 +1,137 @@
+                    responseHeaders.ContentRange = new ContentRangeHeaderValue(size);
+                    responseHeaders.ContentLength = 0;
+                    return (false, start, null);
+                }
+
+                context.Response.StatusCode = (int)HttpStatusCode.PartialContent;
+                responseHeaders.ContentRange = ComputeContentRange(range, size, out start, out var length);
+                responseHeaders.ContentLength = length;
+                return (true, start, length);
+            }
+            else
+            {
+                responseHeaders.ContentLength = size;
+                return (true, 0, null);
+            }
+        }
+
+        private static (bool IsRangeRequest, RangeItemHeaderValue Range) ParseRange(HttpContext context, long length)
+        {
+            var rawRangeHeader = context.Request.Headers.Range;
+            if (StringValues.IsNullOrEmpty(rawRangeHeader))
+            {
+                return (false, null);
+            }
+
+            // Perf: Check for a single entry before parsing it
+            if (rawRangeHeader.Count > 1 || (rawRangeHeader[0] ?? string.Empty).Contains(','))
+            {
+                // The spec allows for multiple ranges but we choose not to support them because the client may request
+                // very strange ranges (e.g. each byte separately, overlapping ranges, etc.) that could negatively
+                // impact the server. Ignore the header and serve the response normally.
+                return (false, null);
+            }
+
+            var rangeHeader = context.Request.GetTypedHeaders().Range;
+            if (rangeHeader == null)
+            {
+                // Invalid
+                return (false, null);
+            }
+
+            // Already verified above
+            Debug.Assert(rangeHeader.Ranges.Count == 1);
+
+            var ranges = rangeHeader.Ranges;
+            if (ranges == null)
+            {
+                return (false, null);
+            }
+
+            if (ranges.Count == 0)
+            {
+                return (true, null);
+            }
+
+            if (length == 0)
+            {
+                return (true, null);
+            }
+
+            // Normalize the ranges
+            var range = NormalizeRange(ranges.Single(), length);
+
+            // Return the single range
+            return (true, range);
+        }
+
+
+        // Note: This assumes ranges have been normalized to absolute byte offsets.
+        private static ContentRangeHeaderValue ComputeContentRange(RangeItemHeaderValue range, long size, out long start, out long length)
+        {
+            start = range.From!.Value;
+            var end = range.To!.Value;
+            length = end - start + 1;
+            return new ContentRangeHeaderValue(start, end, size);
+        }
+
+        private static RangeItemHeaderValue? NormalizeRange(RangeItemHeaderValue range, long length)
```

Comment:
Here you are ignoring invalid values - but doing that silently.
I think it would be better to explicitly reject anything we aren't supporting

Reply:
Same here, we can safely ignore invalid values because we are returning a different status code depending on if the range could be satisfied or not. 
In my opinion, I would rather receive a response that gives me too much data than receive an error. Too much data may result in a degraded experience on the client but an error can lead to crashes.

If you would still prefer an error in this case I can change it though.